src/main/java/MyArrayList.java

```diff
@@ -0,0 +1,335 @@
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Objects;
+import java.util.Spliterator;
+import java.util.function.UnaryOperator;
+
+public class MyArrayList<T> implements List<T> {
+
+  private T[] tempTable;
```

Comment:
why tempTable? Please use something more meaningful. -  this table is not temporary :)